go/arrow/array/concat_test.go

```diff
@@ -568,3 +569,59 @@ func TestConcatDictionaryNullSlots(t *testing.T) {
 
 	assert.Truef(t, array.Equal(actual, expected), "got: %s, expected: %s", actual, expected)
 }
+
+func TestConcatRunLengthEncoded(t *testing.T) {
```

Comment:
Do you want any tests for the overflow cases?

Reply:
That's a good idea, i'll add a test for that